spyder/workers/updates.py

```diff
@@ -95,5 +96,7 @@ def start(self):
         except Exception:
             error_msg = _('Unable to check for updates.')
 
-        self.error = error_msg
-        self.sig_ready.emit()
+        # Don't show dialog when starting up spyder and an error occur
+        if not self.startup or error_msg is None:
```

Comment:
Hmm no, the idea is not to show the dialog when there is an error:

Maybe it will be easy to understand:
`if not (self.startup and error_message is not None):`

Or 
`if self.startup and error_message is not None:
    return`